diff -Nru dkms-2.2.0.3.orig/dkms dkms-2.2.0.3/dkms
--- dkms-2.2.0.3.orig/dkms	2013-04-04 16:57:08.987698428 +0200
+++ dkms-2.2.0.3/dkms	2013-04-13 14:19:14.182300156 +0200
@@ -455,8 +455,11 @@
     [[ $conf ]] && read_conf_file="$conf"
     [[ $3 ]] && read_conf_file="$3"
 
+    [[ ! -r $read_conf_file && -r $dkms_tree/$module/$module_version/build/dkms.conf ]] && \
+	read_conf_file="$dkms_tree/$module/$module_version/build/dkms.conf"
+
     [[ -r $read_conf_file ]] || die 4 $"Could not locate dkms.conf file." \
-	$"File: $conf does not exist."
+	$"File: $read_conf_file does not exist."
 
     [[ $last_mvka = $module/$module_version/$1/$2 && \
 	$last_mvka_conf = $(readlink -f $read_conf_file) ]] && return
@@ -1436,6 +1439,7 @@
 is_module_added() {
     [[ $1 && $2 ]] || return 1
     [[ -d $dkms_tree/$1/$2 ]] || return 2
+    [[ -d $dkms_tree/$1/$2/source/ ]] || return 2
     [[ -L $dkms_tree/$1/$2/source || -d $dkms_tree/$1/$2/source ]];
 }
 
