--- dkms-2.2.0.3/dkms_find-provides.xz_support~	2013-05-24 23:26:35.530418672 +0200
+++ dkms-2.2.0.3/dkms_find-provides	2013-05-24 23:26:50.492630916 +0200
@@ -64,7 +64,7 @@ TMPDIR=$(mktemp -d ${tmp}/dkms-findprovi
 trap "rm -rf $TMPDIR >/dev/null 2>&1"  QUIT EXIT HUP INT TERM
 
 modlist=
-for cand in $(grep -E '(/lib/modules/.+\.ko$|tgz$|tbz$|tar\.(gz|bz2)$)') $*; do
+for cand in $(grep -E '(/lib/modules/.+\.ko$|tgz$|tbz$|tar\.(gz|bz2|xz)$)') $*; do
     if echo $cand | grep -q -E '/lib/modules/.+\.ko$' > /dev/null 2>&1; then
         modlist="$modlist $cand"
     fi
@@ -76,6 +76,8 @@ for cand in $(grep -E '(/lib/modules/.+\
         opts=${opts}z
     elif bzip2 -t $cand >/dev/null 2>&1; then
         opts=${opts}j
+    elif xz -t $cand >/dev/null 2>&1; then
+        opts=${opts}J
     fi
     tar ${opts}f $cand -C $TMPDIR > /dev/null 2>&1
 done
--- dkms-2.2.0.3/dkms.xz_support~	2013-05-24 23:26:35.529418658 +0200
+++ dkms-2.2.0.3/dkms	2013-05-24 23:32:42.750627787 +0200
@@ -156,7 +156,10 @@ set_module_suffix()
     grep -q '\.gz:' /lib/modules/$kernel_test/modules.dep 2>/dev/null && gz_modules=1
     if [ -n "$gz_modules" ]; then
 	[ -z "$delayed_gzip" ] && module_suffix=${module_suffix}.gz
-   fi
+    fi
+    if [ -n "$xz_modules" ]; then
+	module_suffix=${module_suffix}.xz
+    fi
 }
 
 set_kernel_source_dir()
@@ -244,6 +247,10 @@ do_depmod()
     else
 	/sbin/depmod -a "$1"
     fi
+    grep -q '\.xz:' /lib/modules/$kernel_test/modules.dep 2>/dev/null && xz_modules=1
+    if [ -n "$xz_modules" ]; then
+	[ -z "$delayed_xz" ] && module_suffix=${module_suffix}.xz
+    fi
 }
 
 # This function is a little hairy -- every distro has slightly different tools
@@ -1308,6 +1315,10 @@ do_build()
 	if [ -n "$gz_modules" ]; then
 	    gzip -9f $dkms_tree/$module/$module_version/${kernelver_array[0]}/${arch_array[0]}/module/${dest_module_name[$count]}$module_suffix
 	fi
+	if [ -n "$xz_modules" ]; then
+	    xz -f $dkms_tree/$module/$module_version/${kernelver_array[0]}/${arch_array[0]}/module/${dest_module_name[$count]}$module_suffix
+	fi
+
     # Run the post_build script
     run_build_script post_build "$post_build"
 }
@@ -2602,7 +2613,7 @@ load_tarball()
        fi
     fi
 
-    # Figure out what kind of archive it is (tar.gz, tar, tar.bz, etc)
+    # Figure out what kind of archive it is (tar.gz, tar, tar.bz, tar.xz. etc)
     # Note that this does not depend on the extensions being correct.
     local tar_options=""
     for xpand in gzip bzip xz; do
@@ -2816,6 +2827,9 @@ make_rpm()
 	die 1 $"rpmbuild not present." \
 	    $"Install the rpm-build package."
     fi
+    if `xz -t $archive_location 2>/dev/null`; then
+	tar_options="${tar_options}J"
+    fi
 
     # Read the conf file
     read_conf_or_die "$kernelver" "$arch"
@@ -3589,7 +3603,7 @@ for action_to_run in $action; do
 	match)        check_root && have_one_kernel && run_match;;
 	uninstall)    check_root && have_one_kernel && uninstall_module;;
 	mkdriverdisk) check_root && make_driver_disk;;
-	build)        delayed_gzip=1 && build_modules;;
+	build)        delayed_xz=1 && delayed_gzip=1 && build_modules;;
 	add)          add_module;;
 	mktarball)    make_tarball;;
 	mkrpm)        make_rpm;;
