--- dkms-2.6.1/dkms	2018-08-10 16:58:52.603299345 +0100
+++ dkms-2.6.1/dkms.new	2018-08-10 17:01:35.379330725 +0100
@@ -33,8 +33,9 @@
    DEST_MODULE_NAME dest_module_name MODULES_CONF_OBSOLETES
    DEST_MODULE_LOCATION dest_module_location
    modules_conf_obsoletes MODULES_CONF_ALIAS_TYPE
-   modules_conf_alias_type STRIP strip MODULES_CONF_OBSOLETE_ONLY
-   modules_conf_obsolete_only AUTOINSTALL NO_WEAK_MODULES"
+   modules_conf_alias_type MODULES_CONF_EXTRACT_ALIASES MODULES_CONF_OBSOLETE_ONLY
+   STRIP strip MODULES_CONF_OBSOLETE_ONLY modules_conf_obsolete_only AUTOINSTALL"
+
 
 # Some important regular expressions.  Requires bash 3 or above.
 # Any poor souls still running bash 2 or older really need an upgrade.
@@ -605,6 +606,7 @@
         dest_module_location[$index]=${DEST_MODULE_LOCATION[$index]}
         modules_conf_obsoletes[$index]=${MODULES_CONF_OBSOLETES[$index]}
         modules_conf_alias_type[$index]=${MODULES_CONF_ALIAS_TYPE[$index]}
+        modules_conf_extract_aliases[$index]=${MODULES_CONF_EXTRACT_ALIASES[$index]}        
         case ${MODULES_CONF_OBSOLETE_ONLY[$index]} in
             [yY]*)
                 modules_conf_obsolete_only[$index]="yes"
@@ -998,6 +1000,16 @@
         done
     done
 
+    if [ -d /etc/modprobe.d ]; then
+	local index=0
+	while [ $index -lt ${#dest_module_name[@]} ]; do
+	    if [ "${modules_conf_extract_aliases[$index]}" = yes -a -f $install_tree/${kernelver_array[0]}${dest_module_location[$index]}/${dest_module_name[$index]}$module_suffix ]; then
+		modinfo $install_tree/${kernelver_array[0]}${dest_module_location[$index]}/${dest_module_name[$index]}$module_suffix | sed -n -e "s/alias:\(.*\)$/alias \1 ${dest_module_name[$index]}/p" > /etc/modprobe.d/dkms-aliases-${dest_module_name[$index]}
+	    fi
+	    index=$(($index+1))
+	done
+    fi
+
     # Delete the temp dir
     rm -rf $temp_dir_name
 }
@@ -1846,6 +1858,14 @@
     # Run the post_remove script
     run_build_script post_remove "$post_remove"
 
+    if ! $0 status -m $module -v $module_version | grep -q "installed"; then
+	local index=0
+	while [ $index -lt ${#built_module_name[@]} ]; do
+	    rm -f /etc/modprobe.d/dkms-aliases-${dest_module_name[$index]}
+	    index=$(($index+1))
+	done
+    fi
+
     # Run depmod because we changed /lib/modules
     invoke_command "do_depmod $1" "depmod" background
 
