diff -p -up dkms-2.0.19/dkms.binary_only dkms-2.0.19/dkms
--- dkms-2.0.19/dkms.binary_only	2008-04-02 22:57:50.000000000 +0200
+++ dkms-2.0.19/dkms	2008-04-02 22:59:31.000000000 +0200
@@ -2942,12 +2942,12 @@ make_rpm()
 	    local module_release="${module_version##*-}"
 	    local module_version="${module_version%-*}"
     fi
-    invoke_command "LC_ALL=C rpmbuild --define \"_topdir ${temp_dir_name}\" --define \"version $module_version\" ${module_release:+--define \"rel $module_release\"} --define \"module_name $module\" --define \"kernel_versions ${kernelver_array[*]}\" --define \"mktarball_line $mktarball_line\" --define \"__find_provides  /usr/lib/dkms/find-provides\" --define \"_use_internal_dependency_generator 0\" -ba ${temp_dir_name}/SPECS/dkms_mkrpm.spec > ${temp_dir_name}/rpmbuild.log 2>&1" "rpmbuild"
+    invoke_command "LC_ALL=C rpmbuild --define \"_topdir ${temp_dir_name}\" --define \"version $module_version\" ${module_release:+--define \"rel $module_release\"} --define \"module_name $module\" --define \"kernel_versions ${kernelver_array[*]}\" --define \"mktarball_line $mktarball_line\" --define \"__find_provides  /usr/lib/dkms/find-provides\" --define \"_use_internal_dependency_generator 0\" -bb ${temp_dir_name}/SPECS/dkms_mkrpm.spec > ${temp_dir_name}/rpmbuild.log 2>&1" "rpmbuild"
     if [ "$?" -eq 0 ]; then
 	mkdir -p ${rpm_basedir}
-	cp -a ${temp_dir_name}/SRPMS/* ${temp_dir_name}/RPMS/*/* ${rpm_basedir}/
+	cp -a ${temp_dir_name}/RPMS/*/* ${rpm_basedir}/
 	echo $""
-	cat ${temp_dir_name}/rpmbuild.log | grep ^Wrote | sed -e "s:${temp_dir_name}/:${rpm_basedir}/:" -e 's:SRPMS/::' -e 's:RPMS/.*/::'
+	cat ${temp_dir_name}/rpmbuild.log | grep ^Wrote | sed -e "s:${temp_dir_name}/:${rpm_basedir}/:" -e 's:RPMS/.*/::'
 	echo $""
         echo $"DKMS: mkrpm Completed."
     else
