Index: dkms-2.0.19/dkms
===================================================================
--- dkms-2.0.19/dkms
+++ dkms-2.0.19/dkms	2008-04-18 02:54:48.000000000 +0300
@@ -605,6 +605,8 @@
     # $1 = kernel_version
     # $2 = arch
 
+    [ -n "$force" ] && return
+
     local lib_tree="$install_tree/$1"
     local count=0
     echo $"Running module version sanity check."
@@ -632,7 +634,7 @@
 	    fi
 
 	    if [ -n "$kernels_ver_checksum" -a -n "$dkms_ver_checksum" -a \
-		    "$kernels_ver_checksum" == "$dkms_ver_checksum" -a -z "$force" ]; then
+		    "$kernels_ver_checksum" == "$dkms_ver_checksum" ]; then
 		echo $"" >&2
 		echo $"Good news! Module version $dkms_ver_value for ${dest_module_name[$count]}$module_suffix" >&2
 		echo $"exactly matches what is already found in kernel $1." >&2
@@ -643,7 +645,7 @@
 
 	    if [ -n "$kernels_ver_value" -a -n "$dkms_ver_value" ]; then
 		if [[ ! ( $(VER $dkms_ver_value) > \
-		          $(VER $kernels_ver_value) ) && -z "$force" ]]; then
+		          $(VER $kernels_ver_value) ) ]]; then
 		    echo $"" >&2
 		    echo $"Error! Module version $dkms_ver_value for ${dest_module_name[$count]}$module_suffix" >&2
 		    echo $"is not newer than what is already found in kernel $1 ($kernels_ver_value)." >&2
