--- dkms-2.6.1/dkms	2018-08-10 20:13:48.188066453 +0100
+++ dkms-2.6.1/dkms.symvers~	2018-08-10 20:22:13.950487577 +0100
@@ -1380,6 +1380,11 @@
     echo $"Building module:"
 
     invoke_command "$clean" "cleaning build area" background
+    zcat $dkms_tree/*/kernel-${kernelver_array[0]}-${arch_array[0]}/module/symvers.gz > Module.symvers 2>/dev/null
+    while [ "$count" -lt "${#built_module_name[@]}" ]; do
+        zcat $dkms_tree/*/kernel-${kernelver_array[0]}-${arch_array[0]}/module/symvers.gz > "${built_module_location[$count]}/Module.symvers" 2>/dev/null
+	count=$(($count+1))
+    done    
     echo $"DKMS make.log for $module-$module_version for kernel $kernelver ($arch)" >> "$dkms_tree/$module/$module_version/build/make.log"
     date >> "$dkms_tree/$module/$module_version/build/make.log"
     local the_make_command="${make_command/#make/make -j$parallel_jobs KERNELRELEASE=$kernelver}"
@@ -1405,6 +1410,7 @@
 
     # Save a copy of the new module
     mkdir "$base_dir/module" >/dev/null
+    echo -n | gzip -c > "$dkms_tree/$module/$module_version/${kernelver_array[0]}/${arch_array[0]}/module/symvers.gz"
     for ((count=0; count < ${#built_module_name[@]}; count++)); do
     [[ ${strip[$count]} != no ]] && strip -g "$dkms_tree/$module/$module_version/build/${built_module_location[$count]}${built_module_name[$count]}$module_uncompressed_suffix"
     if [ "$module_compressed_suffix" = ".gz" ]; then
