--- dkms-2.2.0.3/dkms.autoalias	2013-05-24 22:38:12.164909302 +0200
+++ dkms-2.2.0.3/dkms	2013-05-24 22:41:20.267571240 +0200
@@ -32,8 +32,8 @@ readonly dkms_conf_variables="CLEAN REMA
        DEST_MODULE_NAME dest_module_name MODULES_CONF_OBSOLETES
        DEST_MODULE_LOCATION dest_module_location
        modules_conf_obsoletes MODULES_CONF_ALIAS_TYPE
-       modules_conf_alias_type STRIP strip MODULES_CONF_OBSOLETE_ONLY
-       modules_conf_obsolete_only AUTOINSTALL"
+       modules_conf_alias_type MODULES_CONF_EXTRACT_ALIASES MODULES_CONF_OBSOLETE_ONLY
+       STRIP strip MODULES_CONF_OBSOLETE_ONLY modules_conf_obsolete_only AUTOINSTALL"
 
 # Some important regular expressions.  Requires bash 3 or above.
 # Any poor souls still running bash 2 or older really need an upgrade.
@@ -517,6 +517,7 @@ read_conf()
 	dest_module_location[$index]=${DEST_MODULE_LOCATION[$index]}
 	modules_conf_obsoletes[$index]=${MODULES_CONF_OBSOLETES[$index]}
 	modules_conf_alias_type[$index]=${MODULES_CONF_ALIAS_TYPE[$index]}
+	modules_conf_extract_aliases[$index]=${MODULES_CONF_EXTRACT_ALIASES[$index]}
 	case ${MODULES_CONF_OBSOLETE_ONLY[$index]} in
 	    [yY]*) modules_conf_obsolete_only[$index]="yes";;
 	esac
@@ -897,6 +898,16 @@ moduleconfig_remove()
 	done
     done
 
+    if [ -d /etc/modprobe.d ]; then
+	local index=0
+	while [ $index -lt ${#dest_module_name[@]} ]; do
+	    if [ "${modules_conf_extract_aliases[$index]}" = yes -a -f $install_tree/${kernelver_array[0]}${dest_module_location[$index]}/${dest_module_name[$index]}$module_suffix ]; then
+		modinfo $install_tree/${kernelver_array[0]}${dest_module_location[$index]}/${dest_module_name[$index]}$module_suffix | sed -n -e "s/alias:\(.*\)$/alias \1 ${dest_module_name[$index]}/p" > /etc/modprobe.d/dkms-aliases-${dest_module_name[$index]}
+	    fi
+	    index=$(($index+1))
+	done
+    fi
+
     # Delete the temp dir
     rm -rf $temp_dir_name
 }
@@ -1671,6 +1682,14 @@ do_uninstall()
     # Run the post_remove script
     run_build_script post_remove "$post_remove"
 
+    if ! $0 status -m $module -v $module_version | grep -q "installed"; then
+	local index=0
+	while [ $index -lt ${#built_module_name[@]} ]; do
+	    rm -f /etc/modprobe.d/dkms-aliases-${dest_module_name[$index]}
+	    index=$(($index+1))
+	done
+    fi
+
     # Run depmod because we changed /lib/modules
     invoke_command "do_depmod $1" "depmod" background
 
