diff -p -up dkms-2.0.19/dkms.weak_module_name dkms-2.0.19/dkms
--- dkms-2.0.19/dkms.weak_module_name	2008-04-01 22:48:35.000000000 +0200
+++ dkms-2.0.19/dkms	2008-04-01 22:48:53.000000000 +0200
@@ -1754,7 +1754,10 @@ function find_module_from_ko()
 
     for kernellink in $(find $dkms_tree -maxdepth $depth -mindepth $depth -name kernel-\* -type l); do
 	module=$(echo "$kernellink" | awk -F / '{print $(NF-1)}')
-	[ -e "$kernellink/module/${basename_ko}" ] && echo "$module"
+	if [ -e "$kernellink/module/${basename_ko}" ]; then
+	    echo "$module"
+	    return
+	fi
     done
     echo ""
 }
