--- dkms-2.6.1/dkms	2018-08-12 15:43:48.815023126 +0100
+++ dkms-2.6.1/dkms.parallel_fix~	2018-08-12 16:13:02.890385543 +0100
@@ -693,10 +693,12 @@
 
     # Use the generic make and make clean commands if not specified
     if [[ $(VER $1) < $(VER 2.6.6) ]]; then
-        [[ ! $make_command ]] && make_command="make -C $kernel_source_dir SUBDIRS=$dkms_tree/$module/$module_version/build modules"
+         [[ ! $make_command || $make_cooand == " $make_flags" ]] && \ 
+        make_command="make $nake_flags -C $kernel_source_dir SUBDIRS=$dkms_tree/$module/$module_version/build modules"
         [[ ! $clean ]] && clean="make -C $kernel_source_dir SUBDIRS=$dkms_tree/$module/$module_version/build clean"
     else
-        [[ ! $make_command ]] && make_command="make -C $kernel_source_dir M=$dkms_tree/$module/$module_version/build"
+         [[ ! $make_command || $make_cooand == " $make_flags" ]] && \
+        make_command="make  $make_flags -C $kernel_source_dir M=$dkms_tree/$module/$module_version/build"
         [[ ! $clean ]] && clean="make -C $kernel_source_dir M=$dkms_tree/$module/$module_version/build clean"
     fi
 
