diff -p -up dkms-2.0.19/dkms.autoload_instead_of_udevadm dkms-2.0.19/dkms
--- dkms-2.0.19/dkms.autoload_instead_of_udevadm	2010-05-04 17:25:36.000000000 +0200
+++ dkms-2.0.19/dkms	2010-05-04 17:26:09.000000000 +0200
@@ -1270,11 +1270,11 @@ function install_module()
 	fi
     fi
 
-    # Notify udev if we installed something for the currently running kernel
-    if [ -x /sbin/udevadm -a "${kernelver_array[0]}" == "$(uname -r)" -a "${arch_array[0]}" == "$(uname -m)" ]; then
-       /sbin/udevadm trigger
-    elif [ -x /sbin/udevtrigger -a "${kernelver_array[0]}" == "$(uname -r)" -a "${arch_array[0]}" == "$(uname -m)" ]; then
-	/sbin/udevtrigger
+    # Load installed modules
+    current_arch=`uname -m`
+    [[ $current_arch = i?86 ]] && current_arch=i586
+    if [ "${kernelver_array[0]}" == "$(uname -r)" -a "${arch_array[0]}" == "$current_arch" ]; then
+        /usr/sbin/dkms_autoload ${dest_module_name[@]}
     fi
 
     echo $""
