diff -up dkms-2.8.3/dkms.13~ dkms-2.8.3/dkms
--- dkms-2.8.3/dkms.13~	2020-08-11 21:54:40.615608428 +0200
+++ dkms-2.8.3/dkms	2020-08-11 21:55:32.548449127 +0200
@@ -1512,6 +1512,13 @@ sign_build()
     fi
 }
 
+display_plymouth_message()
+{
+    if [ -x /bin/plymouth ] && /bin/plymouth --ping; then
+	/bin/plymouth display-message --text="$1"
+    fi
+}
+
 build_module()
 {
     prepare_build
@@ -1788,7 +1795,9 @@ maybe_build_module() (
         echo $"Module $1/$2 already built for kernel $3/4"
         return 0
     }
+    display_plymouth_message "Building $module driver. This may take some time..."
     module="$1" module_version="$2" kernelver="$3" arch="$4" build_module
+    display_plymouth_message ""
 )
 
 maybe_install_module() (
@@ -1796,7 +1805,9 @@ maybe_install_module() (
         echo $"Module $1/$2 already installed on kernel $3/$4"
         return 0
     }
+    display_plymouth_message "Installing $module driver..."
     module="$1" module_version="$2" kernelver="$3" arch="$4" install_module
+    display_plymouth_message ""
 )
 
 build_modules() {
