diff -p -up dkms-2.0.19/dkms.mdkrpm-split-ver-rel dkms-2.0.19/dkms
--- dkms-2.0.19/dkms.mdkrpm-split-ver-rel	2008-04-01 18:28:53.000000000 +0200
+++ dkms-2.0.19/dkms	2008-04-01 18:31:54.000000000 +0200
@@ -2935,7 +2935,11 @@ make_rpm()
     trap 'rm -rf $temp_dir_name' EXIT HUP TERM
     mkdir -p ${temp_dir_name}/{BUILD,RPMS,SRPMS,SPECS,SOURCES}
     cp ${SPECFILE} ${temp_dir_name}/SPECS/dkms_mkrpm.spec
-    invoke_command "LC_ALL=C rpmbuild --define \"_topdir ${temp_dir_name}\" --define \"version $module_version\" --define \"module_name $module\" --define \"kernel_versions ${kernelver_array[*]}\" --define \"mktarball_line $mktarball_line\" --define \"__find_provides  /usr/lib/dkms/find-provides\" --define \"_use_internal_dependency_generator 0\" -ba ${temp_dir_name}/SPECS/dkms_mkrpm.spec > ${temp_dir_name}/rpmbuild.log 2>&1" "rpmbuild"
+    if [ "${module_version%-*}" != "${module_version##*-}" ]; then
+	    local module_release="${module_version##*-}"
+	    local module_version="${module_version%-*}"
+    fi
+    invoke_command "LC_ALL=C rpmbuild --define \"_topdir ${temp_dir_name}\" --define \"version $module_version\" ${module_release:+--define \"rel $module_release\"} --define \"module_name $module\" --define \"kernel_versions ${kernelver_array[*]}\" --define \"mktarball_line $mktarball_line\" --define \"__find_provides  /usr/lib/dkms/find-provides\" --define \"_use_internal_dependency_generator 0\" -ba ${temp_dir_name}/SPECS/dkms_mkrpm.spec > ${temp_dir_name}/rpmbuild.log 2>&1" "rpmbuild"
     if [ "$?" -eq 0 ]; then
 	mkdir -p ${rpm_basedir}
 	cp -a ${temp_dir_name}/SRPMS/* ${temp_dir_name}/RPMS/*/* ${rpm_basedir}/
