Index: dkms-2.0.17.6/dkms_autoinstaller
===================================================================
--- dkms-2.0.17.6/dkms_autoinstaller
+++ dkms-2.0.17.6/dkms_autoinstaller	2008-03-06 23:01:29.000000000 +0200
@@ -72,7 +72,7 @@
 		already_installed=""
 		already_installed_version=""
 		for filename in `ls "$dkms_tree/$module_in_tree"`; do
-		    if [ -d "$dkms_tree/$module_in_tree/$filename" ] && ! [ -h "$dkms_tree/$module_in_tree/$filename" ] && [ "$filename" != "original_module" ]; then
+		    if [ -d "$dkms_tree/$module_in_tree/$filename" ] && ! [ -h "$dkms_tree/$module_in_tree/$filename" ] && [ "$filename" != "original_module" ] && [ -e "$dkms_tree/$module_in_tree/$filename/source/dkms.conf" ]; then
 			version_count=$(($version_count + 1))
 			version_in_tree="$filename"
 
